pbrEngine/engine.py

```diff
@@ -691,6 +749,291 @@ def _resetBlur(self):
+        """
+        self._enableSong(Locations.SONG_FANFARE_COMPLETED, "sound/ME_Fin.brstm")
+
+    def _enableColosseumSelectionMusic(self):
+        """
+        Enables the music for the Colosseum Selection screen. (System01)
+        """
+        self._enableSong(Locations.SONG_COLOSSEUM_SELECTION, "sound/System01.brstm")
+
+    def _enableReceptionMusic(self):
+        """
+        Enables the music for the Reception screen. (System03)
+        """
+        self._enableSong(Locations.SONG_RECEPTION, "sound/System03.brstm")
+
+    def _enableMonSelectionMusic(self):
+        """
+        Enables the music for the Pokemon Selection screen. (System05)
+        """
+        self._enableSong(Locations.SONG_MON_SELECTION, "sound/System05.brstm")
+
+    def _enableControlsMusic(self):
+        """
+        Enables the music for the Controls screen. (System09)
+        """
+        self._enableSong(Locations.SONG_CONTROLS, "sound/System09.brstm")
+
+    def _enableBossMusic(self):
+        path = ""
+        side = None
+        if self.avatars["blue"]["APPEARANCE"]["CHARACTER_STYLE"] > 6:
+            side = "blue"
+        elif self.avatars["red"]["APPEARANCE"]["CHARACTER_STYLE"] > 6:
+            side = "red"
+        if not side:
+            return
+        if self.avatars[side]["APPEARANCE"]["CHARACTER_STYLE"] in [7, 8]:
+            path = "sound/Battleboss01.brstm"
+        elif self.avatars[side]["APPEARANCE"]["CHARACTER_STYLE"] == 9:
+            path = "sound/Battleboss03.brstm"
+        elif self.avatars[side]["APPEARANCE"]["CHARACTER_STYLE"] == 10:
+            path = "sound/Battleboss04.brstm"
+        elif self.avatars[side]["APPEARANCE"]["CHARACTER_STYLE"] == 11:
+            path = "sound/Battleboss02.brstm"
+        elif self.avatars[side]["APPEARANCE"]["CHARACTER_STYLE"] == 12:
+            path = "sound/Battlelastboss.brstm"
+
+        addrs = [Locations.SONG_WATERFALL, Locations.SONG_MAGMA, Locations.SONG_COURTYARD, Locations.SONG_SUNNY_PARK,
+                 Locations.SONG_SUNSET, Locations.SONG_STARGAZER, Locations.SONG_GATEWAY, Locations.SONG_NEON,
+                 Locations.SONG_MAIN_STREET, Locations.SONG_CRYSTAL, Locations.SONG_LAGOON]
+        for addr in addrs:
+            self._disableSong(addr, 8)
+            self._enableSong(addr, path)
+
+        self._disableSong(Locations.SONG_FANFARE_VAR1, 7)  # set boss fanfares
+        self._enableSong(Locations.SONG_FANFARE_VAR1, "sound/ME_Fan02.brstm")
+        self._disableSong(Locations.SONG_FANFARE_VAR3, 7)
+        self._enableSong(Locations.SONG_FANFARE_VAR3, "sound/ME_Fan04.brstm")
+        self._disableSong(Locations.SONG_FANFARE_VAR5, 7)
+        self._enableSong(Locations.SONG_FANFARE_VAR5, "sonnd/ME_Fan06.brstm")
+
+    def _disableSong(self, loc, count):
+        writes = []
+        for i in range(count):
+            writes.append((32, loc.value.addr + 0x4 * i, 0))
+        self._dolphinIO.writeMulti(writes)
```

Comment:
oh, those magic numbers are "count"? please pass the arguments as keywords for clarity

Reply:
why 0x4? what does the loop do? short explanation